### PR TITLE
Added substream sink postgres

### PR DESCRIPTION
**Pull request by harshrjjpt**
### Created a substream sink postgres to populate ethereum network data to postgres

**add_block_entity**:

This function creates a block entity in the block_meta table by extracting relevant information from the given eth::Block object.
It sets various properties such as block ID, number, parent hash, timestamp, size, nonce, receipt root, gas limit, and gas used.

**add_trx_info_entity**:

This function creates a transaction entity in the transactions table using the provided eth::TransactionTrace object.
It sets properties such as transaction ID, status, gas used, gas limit, transaction nonce, to address, from address, max fee per gas, max priority fee per gas, block number, and timestamp.

**add_contracts_info_entity**:

This function creates a contract entity in the contracts table based on the given eth::TransactionTrace object.
It sets properties such as contract ID, transaction hash, address, owner, block number, and timestamp.

**db_out**:

This is the main function that processes an eth::Block and generates DatabaseChanges as the result.
It iterates over the transaction traces of the block and calls the appropriate functions to create block, transaction, and contract entities in the tables.
It handles successful transactions by checking the status and processing contracts data.

### Created a schema file to store tables 

**eth.block_meta:**
This table stores information about blocks in the Ethereum network.

**eth.cursors:**
This table manages cursors for tracking the progress of blockchain processing.

**eth.transactions:**
This table stores information about transactions in the Ethereum network.

**eth.contracts:**
This table stores information about contracts in the Ethereum network.

### CODE REVIEW

File: substreams-sink-postgres/docs/substream/.gitignore
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/.gitignore:1]: Don't need this file inside folder.

File: substreams-sink-postgres/docs/substream/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+base64 = "0.13.0"
+hex = "0.4"
+log = "0.4.14"
+prost-types = "0.11"
+chrono = { version = "0.4", features = [ "std" ], default-features = false }
+
+[profile.release]
+lto = true
+opt-level = 's'
+strip = "debuginfo"
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/Cargo.toml]: Add end of line

File: substreams-sink-postgres/docs/substream/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+substreams = "0.5.0"
+substreams-ethereum = "0.8.0"
+prost = "0.11"
+substreams-database-change = "1.2.0"
+substreams-entity-change = {git = "https://github.com/streamingfast/substreams-entity-change/", branch = "develop"}
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/Cargo.toml]: What we need to have streamingfast github link?

File: substreams-sink-postgres/docs/substream/Makefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+stream: build
+	substreams run -e mainnet.eth.streamingfast.io:443 substreams.yaml db_out -s 12292922 -t +10
+
+.PHONY: stream_db
+stream_db: build
+	substreams run -e $(ENDPOINT) substreams.yaml db_out -t $(STOP_BLOCK)
+
+.PHONY: protogen
+protogen:
+	substreams protogen ./substreams.yaml --exclude-paths="sf/substreams,google"
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/Makefile]: End of file

File: substreams-sink-postgres/docs/substream/README.md
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/README.md:1]: Do we need this?

File: substreams-sink-postgres/docs/substream/proto/block_meta.proto [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+// ## Transaction Details ##
+message Transaction {
+  string id = 1;
+  string status = 2;
+  uint64 gasUsed = 3;
+  uint64 gasLimit = 4;
+  uint64 nonce = 5;
+  string to = 6;
+  string from = 7;
+  // string maxFeePerGas = 8;
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/proto/block_meta.proto]: Remove if not required.

File: substreams-sink-postgres/docs/substream/rust-toolchain.toml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "1.64.0"
+components = [ "rustfmt" ]
+targets = [ "wasm32-unknown-unknown" ]
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/rust-toolchain.toml]: New line at end of file.

File: substreams-sink-postgres/docs/substream/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+) -> Result<DatabaseChanges, substreams::errors::Error> {
+    let header = &blk.header.as_ref().unwrap();
+    let block_number = &blk.number;
+    let time_stamp = header.timestamp.clone().unwrap().seconds;
+    let mut tables = Tables::new();
+    // get blocks data
+    add_block_entity(&mut tables, &blk);
+    for trx in &blk.transaction_traces{
+        // get transactions data
+        log::info!("hello");
```
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/src/lib.rs]: Required this log?

File: substreams-sink-postgres/docs/substream/src/pb/mod.rs
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/src/pb/mod.rs:1]: Remove commented code.

File: substreams-sink-postgres/docs/substream/substreams.yaml
Comment (realchoubey) [inline substreams-sink-postgres/docs/substream/substreams.yaml:1]: Remove commented code.

File: substreams-sink-postgres/schemas /.gitignore
Comment (realchoubey) [inline substreams-sink-postgres/schemas /.gitignore:1]: .gitignore file should be at root of repo.

File: substreams-sink-postgres/schemas /eth-block-meta/schema.sql [hunk 7]
```diff
@@ -0,0 +1,49 @@
+CREATE SCHEMA eth;
+
+CREATE TABLE eth.block_meta (
```
Comment (realchoubey) [inline substreams-sink-postgres/schemas /eth-block-meta/schema.sql]: Can we change the name of this table? meta mostly used for basic details of something, here we are saving all details of block

**Review by realchoubey — CHANGES_REQUESTED** [review]
It seems like `schemas` folder have extra space at the end, please confirm and correct it.
Many files don't have extra line at the end of file

File: substreams-sink-postgres/schemas /eth-block-meta/schema.sql (hunk 7 above)
Comment (harshrjjpt) [reply substreams-sink-postgres/schemas /eth-block-meta/schema.sql]: block_details?

**Pull request merged**